RoosterPlanner.Api/Controllers/AvailabilityController.cs

```diff
@@ -0,0 +1,195 @@
+using RoosterPlanner.Api.Models;
+using RoosterPlanner.Api.Models.Constants;
+using RoosterPlanner.Models;
+using RoosterPlanner.Models.Models.Enums;
+using RoosterPlanner.Models.Types;
+using RoosterPlanner.Service;
+using RoosterPlanner.Service.DataModels;
+using RoosterPlanner.Service.Helpers;
+using Type = RoosterPlanner.Api.Models.Type;
+
+namespace RoosterPlanner.Api.Controllers
+{
+    [Authorize]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AvailabilityController : ControllerBase
+    {
+        private readonly ILogger<AvailabilityController> logger;
+        private readonly IAvailabilityService availabilityService;
+        private readonly IParticipationService participationService;
+        private readonly IShiftService shiftService;
+        private readonly IProjectService projectService;
+        private readonly ITaskService taskService;
+
+        public AvailabilityController(
+            ILogger<AvailabilityController> logger,
+            IAvailabilityService availabilityService,
+            IParticipationService participationService,
+            IProjectService projectService,
+            IShiftService shiftService,
+            ITaskService taskService)
+        {
+            this.logger = logger;
+            this.availabilityService = availabilityService;
+            this.shiftService = shiftService;
+            this.participationService = participationService;
+            this.projectService = projectService;
+            this.taskService = taskService;
+        }
+
+        [HttpGet("find/{projectId}/{userId}")]
+        public async Task<ActionResult<AvailabilityDataViewModel>> GetAvailabilityData(Guid projectId, Guid userId)
+        {
+            if (projectId == Guid.Empty)
+                return BadRequest("No valid projectId.");
+            if (userId == Guid.Empty)
+                return BadRequest("No valid userId.");
+            try
+            {
+                TaskListResult<ProjectTask> taskResult = await taskService.GetAllProjectTasksAsync(projectId);
+                TaskListResult<Shift> shiftResult =
+                    await shiftService.GetShiftsWithAvailabilitiesAsync(projectId, userId);
+
+                if (!taskResult.Succeeded)
+                    return UnprocessableEntity(new ErrorViewModel
+                        {Type = Type.Error, Message = taskResult.Message});
+                if (!shiftResult.Succeeded)
+                    return UnprocessableEntity(new ErrorViewModel
+                        {Type = Type.Error, Message = shiftResult.Message});
+
+                List<Schedule> knownAvailabilities = new List<Schedule>();
+
+                var x = shiftResult.Data.GroupBy(s => s.Date);
+                foreach (IGrouping<DateTime, Shift> grouping in shiftResult.Data.GroupBy(s => s.Date))
+                {
+                    int numberOfShifts = grouping.Count();
+                    int numberOfAvailabilities = 0;
```

Comment:
Variabele zou je buiten de for-each kunnen initialiseren en hier alleen resetten (op 0 zetten)